src/tests/user/profile.test.ts

```diff
@@ -0,0 +1,95 @@
+import request from 'supertest';
+import { faker } from '@faker-js/faker';
+import jwt from 'jsonwebtoken';
+
+import app from '../../index.js';
+import { AppDataSource } from '../../config/database.js';
+import { User } from '../../entities/User.js';
+import { configs } from '../../config/env.js';
+
+let randomUser: User;
+let validToken: string;
+let expiredToken: string;
+let invalidToken: string;
+let validTokenForNotRegisteredUser: string
+const getUserProfileURL = '/api/users/profile';
+
+beforeAll(async () => {
+  await AppDataSource.initialize();
+
+  const newUser = {
+    email: faker.internet.email(),
+    password: faker.internet.password(),
+    name: faker.person.fullName(),
+  };
+  const savedUser = await AppDataSource.getRepository(User).save(newUser);
```

Comment:
Use the register route here to make sure the password is hashed before saving to database, after user registration, make sure the status is 201, if not throw an error.